provisioning_templates/snippet/redhat_register.erb

```diff
@@ -100,6 +109,31 @@ snippet: true
     fi
   <% end %>
 
+  <%- if (host_param('syspurpose_role') || host_param('syspurpose_usage') || host_param('syspurpose_sla') || host_param('syspurpose_addons')) %>
+    <%- if !atomic %>
+      dnf -y install subscription-manager-syspurpose
+    <%- end %>
+
+    if [ -f /usr/sbin/syspurpose ]; then
+      <%- if host_param('syspurpose_role') %>
+        syspurpose set-role "<%= host_param('syspurpose_role') %>"
+      <%- end %>
+      <%- if host_param('syspurpose_usage') %>
+        syspurpose set-usage "<%= host_param('syspurpose_usage') %>"
+      <%- end %>
+      <%- if host_param('syspurpose_sla') %>
+        syspurpose set-sla "<%= host_param('syspurpose_sla') %>"
+      <%- end %>
+      <%- if host_param('syspurpose_addons') %>
+        <% addons = host_param('syspurpose_addons').split(',')
+            .map { |add_on| "'#{add_on.strip}'" }.join(" ") %>
+        syspurpose add-addons <%= addons %>
+      <%- end %>
+    else
+      echo "Syspurpose CLI not found."
```

Comment:
Would this be an error if they're set but unavailable?

Reply:
Depends, but not really, it could be set on a host group that's provisioning hosts that do and do not support it.

Reply:
We don't really care a whole lot if a host can't get a purpose set, it's just a hint to subscription-manager about what to do.